packages/yarnpkg-core/sources/scriptUtils.ts

```diff
@@ -250,11 +250,20 @@ export async function prepareExternalProject(cwd: PortablePath, outputPath: Port
               ? [`workspace`, workspace]
               : [];
 
+            // `set version` will update the Manifest to contain a `packageManager` field with the latest
+            // Yarn version which causes the results to change depending on when this command was run,
+            // therefore we revert any change made to it.
+            const manifestPath = ppath.join(cwd, Filename.manifest);
+            const manifestBuffer = await xfs.readFilePromise(manifestPath);
+
             // Makes sure that we'll be using Yarn 1.x
             const version = await execUtils.pipevp(process.execPath, [process.argv[1], `set`, `version`, `classic`, `--only-if-needed`], {cwd, env, stdin, stdout, stderr, end: execUtils.EndStrategy.ErrorCode});
             if (version.code !== 0)
               return version.code;
 
+            // Revert any changes made to the Manifest by `set version`.
+            await xfs.writeFilePromise(manifestPath, manifestBuffer);
```

Comment:
We probably should also use `--yarn-path` to make sure that Yarn will set `yarnPath` - in v4 it may not be true anymore, since it prefers using exclusively Corepack if possible.

Reply:
Done